.github/workflows/docker.yml

```diff
@@ -0,0 +1,81 @@
+# GitHub Actions workflow to build and push Docker images.
+name: "Docker"
+on:
+  push:
+    tags: [ "v*.*.*" ]
+
+concurrency:
+  group: "docker-${{ github.event.number || github.ref }}"
+  cancel-in-progress: false
+
+jobs:
+  # Prepare to deploy and release
+  prepare:
+    name: "Prepare"
+    runs-on: "ubuntu-latest"
+    permissions:
+      contents: read
+    outputs:
+      version: "${{ steps.data.outputs.version }}"
+      tag: "${{ steps.data.outputs.tag }}"
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: "Retrieve build data"
+        id: "data"
+        env:
+          RAW_VERSION: "${{ github.ref_name }}"
+        run: |
+          VERSION=$(echo "$RAW_VERSION" | sed -e 's/^v//')
+          TAG=$(echo "$RAW_VERSION" | sed -E 's/^([^v])/v\1/g')
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "tag=$TAG" >> "$GITHUB_OUTPUT"
+
+  # Build and push Docker image
+  docker:
+    name: "Docker (hemiconnector)"
+    uses: hemilabs/actions/.github/workflows/docker.yml@main
```

Comment:
Even when we manage the repo, use a tag or a hash instead of a branch as version reference.

Reply:
Okay, pinned to latest commit on `main`.